Adds module code

ksatirli opened the pull request:
Hello reviewers 👋🏼 

This PR adds the base for this module. I'm specifically interested in docs reviews, to ensure practitioners have a solid base to work with.

im2nguyen commented on README.md:
```diff
@@ -0,0 +1,84 @@
+
+### Unsealing Vault
+
+Once the add-on has been deployed, the Vault server can be unsealed using the following commands.
```
```suggestion
Once the add-on has been deployed, you can unseal the Vault server using the following commands.
```

im2nguyen commented on README.md:
```diff
@@ -0,0 +1,84 @@
+
+> You will need to be in the `vault` (Kubernetes) namespace while running these commands, by default
+
+You will first need to initialize the Vault server:
```
```suggestion
First, initialize the Vault server. This will return [unseal keys](https://www.vaultproject.io/docs/concepts/seal#seal-unseal) and [root token](https://www.vaultproject.io/docs/concepts/tokens#root-tokens). Store this in a safe place since you will use them to unseal the Vault server.
```

im2nguyen commented on README.md:
````diff
@@ -0,0 +1,84 @@
+```sh
+kubectl exec -it vault-0 -n vault -- vault operator init
+```
+
+Take note of the [unseal keys](https://www.vaultproject.io/docs/concepts/seal#seal-unseal) and [root token](https://www.vaultproject.io/docs/concepts/tokens#root-tokens) that get generated.
````
```suggestion
```

im2nguyen commented on README.md:
````diff
@@ -0,0 +1,84 @@
+kubectl exec -it vault-0 -n vault -- vault operator unseal <key 3>
+ ```
+
+Confirm that the Vault server is unsealed by checking the status of the Vault server:
+
+```sh
+kubectl get pods -n vault | grep vault
+
````
````suggestion
Confirm that the Vault server is unsealed by checking the status of the Vault server.

```sh
kubectl get pods -n vault | grep vault
```

This command will return a similar output to the following.

```sh

````

im2nguyen replied:
this way, users can c/p the `kubectl` command by clicking the button

ksatirli replied:
Love it!

im2nguyen commented on README.md:
````diff
@@ -0,0 +1,84 @@
+vault-agent-injector |  1/1  | Running | 0        | 1m
+```
+
+At this point, Vault can be used to store, access and deploy secrets to your application workloads.
````
active voice

```suggestion
At this point, you can use Vault to store, access and deploy secrets to your application workloads.
```

im2nguyen commented on README.md:
```diff
@@ -0,0 +1,84 @@
+
+At this point, Vault can be used to store, access and deploy secrets to your application workloads.
+
+See [this guide](https://learn.hashicorp.com/tutorials/vault/getting-started-first-secret?in=vault/getting-started) for a detailed overview on how to get started.
```
able-ist lang

```suggestion
Refer to [this guide](https://learn.hashicorp.com/tutorials/vault/getting-started-first-secret?in=vault/getting-started) for a detailed overview on how to get started.
```

im2nguyen reviewed (approved):
a couple of clarifications but overall looks great! 🎉 

same suggestions apply to `blueprints/getting-started/README.md`

ksatirli commented:
Thanks @im2nguyen - I've applied your suggestion.

Care for a re-check?

onlydole commented on blueprints/getting-started/variables.tf:
```diff
@@ -0,0 +1,27 @@
+variable "kubernetes_version" {
+  type        = string
+  description = "Kubernetes version to use for EKS Cluster"
+  default     = "1.21"
```
*Nit* - Kubernetes 1.22 is out for EKS and has some nice new features! But this nit isn’t a blocker to merging this in.

ksatirli replied:
Let me run a few tests - if they come back clean, I'll bump to `1.22`

onlydole commented on .github/workflows/terraform-docs.yml:
```diff
@@ -16,7 +16,15 @@ jobs:
           output-method: inject
           git-push: true
+
+      - name: Render documentation for `getting-started` example and push changes back to branch
```
Ohhh - this is a great addition!

ksatirli replied:
Thanks - really enjoyed this one!

onlydole commented on blueprints/getting-started/README.md:
```diff
@@ -0,0 +1,97 @@
+
+These commands may be used to configure a local `vault` agent and intialize the server, as described in the next section.
+
+### Unsealing Vault
```
*Nit* - It may help to reduce some duplication between READMEs and use either the main README or this one, then provide a link between the two for this section (and any others that make sense).

ksatirli replied:
Great catch - discussing this with @im2nguyen now.

onlydole reviewed (approved):
This looks great to me! I haven’t gotten a chance to test a full E2E implementation, but I will in the very near future. 

I left some *nit* comments, but they’re not blockers for merging this in, as they’re only EKS  config updates and docs reshuffling items.

ksatirli commented:
@im2nguyen and @onlydole, thanks for your reviews and suggestions!

I've updated this PR with these changes and cleaned up the READMEs some more.

Care to re-review?

onlydole reviewed (approved):
Fantastic! 🎉 LGTM 👍🏼
